src/spikeinterface/postprocessing/isi.py

```diff
@@ -56,6 +56,26 @@ def _select_extension_data(self, unit_ids):
         new_extension_data = dict(isi_histograms=new_isi_hists, bins=new_bins)
         return new_extension_data
 
+    def _merge_extension_data(
+        self, units_to_merge, new_unit_ids, new_sorting_analyzer, censor_ms=None, verbose=False, **job_kwargs
+    ):
+        new_bins = self.data["bins"]
+        arr = self.data["isi_histograms"]
+        num_dims = arr.shape[1]
+        all_new_units = new_sorting_analyzer.unit_ids
+        new_isi_hists = np.zeros((len(all_new_units), num_dims), dtype=arr.dtype)
+        for unit_ind, unit_id in enumerate(all_new_units):
+            if unit_id not in new_unit_ids:
+                keep_unit_index = self.sorting_analyzer.sorting.id_to_index(unit_id)
+                new_isi_hists[unit_ind] = arr[keep_unit_index]
+            else:
+                new_sorting = new_sorting_analyzer.sorting.select_units([unit_id])
+                new_hist, _ = _compute_isi_histograms(new_sorting, **self.params)
+                new_isi_hists[unit_ind] = new_hist
```

Comment:
I have the intuition that this is sub optimal to make many select_units instead of one.

Reply:
This is really fast, not sure this is worth optimizing